src/eredis_cluster_monitor.erl

```diff
@@ -172,13 +172,13 @@ get_cluster_nodes() ->
     Options = get_current_options(State),
     get_cluster_nodes(State, Options).
 
--spec get_cluster_nodes([#node{}], options()) -> [[bitstring()]].
+-spec get_cluster_nodes(State :: #state{}, Options :: options()) -> [[bitstring()]].
 get_cluster_nodes(State, Options) ->
     Query = ["CLUSTER", "NODES"],
     FailFn = fun(_Node) -> "" end, %% No default data to use when query fails
     ClusterNodes = get_cluster_info(State, Options, Query, FailFn),
     %% Parse result into list of element lists
-    NodesInfoList = binary:split(ClusterNodes, <<"\n">>, [global, trim]),
+    NodesInfoList = binary:split(iolist_to_binary(ClusterNodes), <<"\n">>, [global, trim]),
```

Comment:
Can this ever have worked? Maybe this function is dead code which should be removed?

Reply:
I guess the "problem" was that get_cluster_info/4 spec was set to return the common type iodata(), which is iolist() | binary()  and binary:split only takes binary.
Alternative is to modify the spec get_cluster_info/4 to use a stricter "string" type

It seems to be a mess with naming of the functions:
https://medium.com/erlang-battleground/playing-with-strings-c71b9ebed018

Reply:
`get_cluster_info/4` doesn't always return a binary. In the main flow case, it returns what `eredis:q(Connection, Query)` returns, where Query is "CLUSTER NODES". It's a nested list of some sort.

Yes, the standard library naming of functions is a mess.